pkgs/build-support/go/build-hook.sh

```diff
@@ -0,0 +1,140 @@
+
+goBuildHook() {
```

Comment:
```suggestion
goBuildPhase() {
```

AFAICT, `fooBarHook` is meant to be run when doing `runHook foo`.

Also, I would propose defining `goBuild`, which runs all the `goBuildPhase` commands except `runHook preBuild` and `runHook postBuild`.  This way, `goBuild` could be inserted into the `pre`- and `post`-phases of other build helpers/workflows. For example, `apptainer` uses the `stdenv.mkDerivation`'s Make-based workflow during configuration and build but still requires the Go modules realization from `goConfigurePhase`. If we have `goConfigure`, we could plug it into `preConfigure` and switch on `dontGoConfigure` and `dontGoBuild`.